### ISSUE TITLE
`clippy::obfuscated_if_else` expands macros in suggestion

### ISSUE THREAD

**Issue by akiramusic000** [rust-lang/rust-clippy#16288]
### Summary

The suggestion for the `clippy::obfuscated_if_else` lint ends up expanding macros, which probably shouldn't happen.

### Reproducer

Code:

```rust
#![allow(clippy::disallowed_names)]

fn main() {
    true
        .then(|| format!("this is a test"))
        .unwrap_or_default();
}
```

Currently, the suggestion ends up being:
```rust
if true { $crate::__export::must_use({
    $crate::fmt::format($crate::__export::format_args!($($arg)*))
}) } else { Default::default() };
```
Instead, the suggestion should be:
```rust
if true { format!("this is a test") } else { Default::default() };
```

### Version

```text
rustc 1.94.0-nightly (a6525d526 2025-12-21)
binary: rustc
commit-hash: a6525d5264da34f51ad48c178281d3c6323dbfcf
commit-date: 2025-12-21
host: x86_64-apple-darwin
release: 1.94.0-nightly
LLVM version: 21.1.8
```
